GLOSSARY.md

```diff
@@ -40,7 +40,13 @@ This glossary accompanies the [GitOps Principles](./PRINCIPLES.md), and other su
     Git, from which GitOps derives its name, is the canonical example used as this state store but any other system that meets these criteria may be used.
     In all cases, these state stores must be properly configured and precautions must be taken to comply with requirements set out in the GitOps Principles.
 
+- ## Intermediate State Store
+    A system for storing a copy of the declarations that are mastered in the State Store. This system's purpose is intended to bridge the gap in availability between that of the State Store and the expected availability to make configuration changes to the Software System. The Intermediate State Store will offer an availability the same as or near enough to that of the users' expectations to update configuration in the Software System.
+    Where an Intermediate State Store is used, Reconciliation is used between the State Store and the Intermediate State Store and then again between the Intermediate State Store and the Software System.
```

Comment:
I think this part has gotten (and will become) much more relevant with people adopting OCI also for things other than container images. Or as @monadic put it:

> GitOps is a transaction system with a Git backend and OCI cache

Reply:
Are we moving into _leader election / consensus_ territory?

Reply:
Where do you see leader election or consensus palying a role here? There is exactly one SSoT state store and N intermediate state stores. In simple setups you'll have exactly one intermediate state store. In more complex scenarios you might have more than one (e.g. one per environment). Synchronization always happens unidirectional from state store into the intermediate state stores. The intermediate state stores are independent of each other and don't need to be synchronized laterally.

```
                           +======+         +==============+
                     +---->| iss1 |<--------| gitops agent |
+=============+      |     +======+         +==============+
| state store |------+            
+=============+      |     +======+         +==============+
                     +---->| iss2 |<--------| gitops agent |
                     |     +======+         +==============+
                     |
                     |     +======+         +==============+
                     +---->| iss3 |<--------| gitops agent |
                           +======+         +==============+
```

Reply:
IMHO what's important to point out here is to make clear the requirements the intermediate state store has to satisfy in order for this whole setup to satisfy the GitOps principles:

1. It must version all the artifacts
2. Artifacts, once stored, must be immutable
3. It must retain a version history